packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/List.tsx

```diff
@@ -177,20 +178,45 @@ const handleEnterKeyOnList = (editor: Editor) => {
   }
 };
 
+/**
+ * Common handler for converting a node to a list
+ */
+const handleConvertToList = (editor: Editor, format: Block<'list'>['format']) => {
+  // Get the element to convert
+  const entry = prepareHandleConvert(editor);
+  if (!entry) return;
+  const [element, elementPath] = entry;
+
+  // Explicitly set non-needed attributes to null so that Slate deletes them
+  const { type: _type, children: _children, ...extra } = element;
+  const attributesToClear: Record<string, null> = {};
+  Object.keys(extra).forEach((key) => {
+    attributesToClear[key] = null;
+  });
```

Comment:
It looks like this same thing is done in almost every handleConvertX, maybe some or all of it could be extracted to a function? 

Mainly this part:
```ts
  const { type: _type, children: _children, ...extra } = element;
  const attributesToClear: Record<string, null> = {};
  Object.keys(extra).forEach((key) => {
    attributesToClear[key] = null;
  });
```

I think `Array.reduce` would be appropriate to avoid mutating an object...but up to you.

Reply:
You're right, there was room for a much bigger abstraction here. I didn't think it was possible at first because many fields have custom logic (creating empty blocks below, wrapping list items...), but that logic actually happens after the base conversion stuff, so I extracted that part